unyt/tests/test_unyt_array.py

```diff
@@ -2493,6 +2493,22 @@ def test_delta_degF():
     assert 2 * delta_degF == unyt_quantity(2, "delta_degF")
 
 
+@pytest.mark.parametrize(
+    ("u0", "u1", "uout"),
+    [
+        (degC, degC, delta_degC),
+        (K, K, delta_degC),
```

Comment:
Sorry to be a pain, but Kelvin minus Kelvin should result in Kelvin, similarly Rankine minus Rankine. Maybe the checks in `_difference_units` should check `base_offset`?

Reply:
good call. I updated the test and the implementation